src/App.tsx

```diff
@@ -1,203 +1,283 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import {
+  deleteTodo,
+  getTodos,
+  patchTodo,
+  postTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import cn from 'classnames';
+
+function areAllTodosCompleted(todos: Todo[]) {
+  return todos.every(todo => todo.completed);
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const [flag, setFlag] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
```

Comment:
You use two states for the error - isError & errorMessage.
You can reduce them to a single with something like

```suggestion
  const [errorMessage, setErrorMessage] = useState('');
  const isError = !!errorMessage;
```